### PR TITLE
Update question answering and delete qdrant from code

### PR DESCRIPTION
**Pull request by lickem22**
Reviewer: @suzinyou 

Estimate: 1h30mins

---

## Ticket

Fixes: [AAQ-356](https://idinsight.atlassian.net/browse/AAQ-356?atlOrigin=eyJpIjoiYjMzODNlNTNlZDY0NGE1OWI0ZjFkYjkzZTY5ODI0ZWEiLCJwIjoiaiJ9) & [AAQ-358 ](https://idinsight.atlassian.net/browse/AAQ-358?atlOrigin=eyJpIjoiNzQ2YTViN2RjNzI0NDQxMTg2MDhiMjRjY2MzMWU4YTgiLCJwIjoiaiJ9)

### Goal
As part of the refactoring to use  `pgvector` ([AAQ-348]( https://idinsight.atlassian.net/browse/AAQ-348?atlOrigin=eyJpIjoiMGM4MTU4MDk0NjllNDA1ZDlkMDQ5MjE0ZTBkZDQ5MzkiLCJwIjoiaiJ9) ), the goal of this PR, is to add the changes to the question_answer endpoints to use pgvector instead of qdrant. We can then update the tests and delete qdrant
### Changes
The following changes have been made:
- Replaced `qdrant` with `pgvector` for every `question_answer` endpoint.
- Refactored tests to use pgvector
- refactored validation to use pgvector
- refactored github action flows
- Deleted qdrant from code.

The documentation and deployment has not been changed yet. But will be updated in next PRs. 
## How has this been tested?
This has been tested by thouroughly testing each `question_answer` endpoint. And running the existing tests to make sure everything is working. 

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [x] I have updated the automated tests (if applicable)
- [x] I have updated the requirements (if applicable)
- [ ] I have updated the README file (if applicable)
- [ ] I have updated affected documentation (if applicable)

[AAQ-356]: https://idinsight.atlassian.net/browse/AAQ-356?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AAQ-348]: https://idinsight.atlassian.net/browse/AAQ-348?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core_backend/Makefile [hunk 1]
```diff
@@ -35,9 +35,9 @@ run-tests:
 	python -m pytest -rPQ -m "not rails"
 
 # Test DBs
-setup-test-containers: setup-test-relational-db setup-test-vector-db #setup-alignscore-container
+setup-test-containers: setup-test-relational-db  setup-alignscore-container
```
Comment (suzinyou) [inline core_backend/Makefile:38]:
```suggestion
setup-test-containers: setup-test-relational-db setup-alignscore-container
```

File: core_backend/Makefile [hunk 2]
```diff
@@ -35,9 +35,9 @@ run-tests:
...
 
-teardown-test-containers: stop-test-relational-db stop-test-vector-db #stop-alignscore-container
+teardown-test-containers: stop-test-relational-db  stop-alignscore-container
```
Comment (suzinyou) [inline core_backend/Makefile:40]:
```suggestion
teardown-test-containers: stop-test-relational-db stop-alignscore-container
```

File: core_backend/app/routers/manage_content.py [hunk 3]
```diff
@@ -27,20 +21,6 @@
 logger = setup_logger()
 
 
-class QdrantPayload(BaseModel):
-    """Content payload for qdrant"""
-
-    # Ensure len("*{title}*\n\n{text}") <= 1600
-    content_title: Annotated[str, StringConstraints(max_length=150)]
-    content_text: Annotated[str, StringConstraints(max_length=1446)]
```
Comment (suzinyou) [inline core_backend/app/routers/manage_content.py:35]: We need these length constraints. Can you add them to the ContentDB orm along with the comment?

File: core_backend/app/db/db_models.py [hunk 4]
```diff
@@ -449,6 +449,20 @@ async def get_similar_content(
     return await get_search_results(asession, question_embedding, n_similar)
 
 
+async def get_similar_content_async(
+    asession: AsyncSession,
+    question: UserQueryBase,
+    n_similar: int,
+) -> Dict[int, UserQuerySearchResult]:
```
Comment (suzinyou) [inline core_backend/app/db/db_models.py:456]: nit: should we put asession as the last argument? To be consistent with the functions in the question_answer router?

File: core_backend/tests/api/conftest.py [hunk 5]
```diff
@@ -33,29 +31,44 @@
...
+def db_session() -> pytest.FixtureRequest:
+    """Create a test database session."""
+    session_gen = get_session()
+    session = next(session_gen)
+
+    try:
+        yield session
+    finally:
+        session.rollback()
+        next(session_gen, None)
```
Comment (suzinyou) [inline core_backend/tests/api/conftest.py:43]: Oo. Could you explain how this works? get_session() returns a generator? What does the code for finally statement do?

File: core_backend/validation/validate_retrieval.py [hunk 6]
```diff
@@ -87,12 +81,12 @@ def prepare_content_data(
...
         aws_profile: Union[str, None] = None,
     ) -> pd.DataFrame:
-        """Prepare content data for loading to qdrant collection"""
+        """Prepare content data for loading to content table"""
         df = pd.read_csv(
             content_data_path,
             storage_options=dict(profile=aws_profile),
         )
-        df["content_id"] = [uuid.uuid4() for _ in range(len(df))]
+        df["content_id"] = [i for i in range(len(df))]
```
Comment (suzinyou) [inline core_backend/validation/validate_retrieval.py:89]:
```suggestion
        df["content_id"] = list(range(len(df)))
```
how about this

**Review by suzinyou — APPROVED** [review]
This is awesome!! All the unit tests pass, and APIs work. I have made a just a few requests. Feel free to merge after addressing!

File: core_backend/tests/api/conftest.py (hunk 5 above)
Comment (lickem22) [reply core_backend/tests/api/conftest.py:43]: Yes get_session returns a generator. I wasn't able to use a context manager to build the fixture. What the code does:
- get the session as a generator,
- Get the first element which is the actual sesson
- Then, yield the session for the whole test
- Then `session.rollback()` to undo the changes made in the database 
- Then we release the session with `next(session_gen, None)`

File: core_backend/app/db/db_models.py (hunk 4 above)
Comment (lickem22) [reply core_backend/app/db/db_models.py:456]: Actually there is not consistency accross the codebase. In `db_models.py` `asession` starts first while in 
most routers,  `asession` is last. Let's agreed on what to use. And I will update in next PR.

**Pull request merged**